pkg/blockstorage/vmware/vmware.go

```diff
@@ -570,6 +629,36 @@ func (p *FcdProvider) getSnapshotTags(ctx context.Context, fullSnapshotID string
 
 // SnapshotsList is part of blockstorage.Provider
 func (p *FcdProvider) SnapshotsList(ctx context.Context, tags map[string]string) ([]*blockstorage.Snapshot, error) {
+	filterStr := generateDescription(tags)
+	if filterStr == "" {
+		return p.snapshotsListByTag(ctx, tags)
+	}
```

Comment:
I think the function would flow smoother if the "true" part of the check invoked a description specific method. e..g
```
func (p *FcdProvider) SnapshotsList(ctx context.Context, tags map[string]string) ([]*blockstorage.Snapshot, error) {
  if snapDesc := generateDescription(tags); snapDesc != "" {
    return p.snapshotsListByDescription(ctx, snapDesc, tags)
  }

  return p.snapshotListByTag(ctx, tags)
}

Reply:
The `p.snapshotListByDescription` function could be used directly by `getCreatedSnapshot`